mesonbuild/backend/ninjabackend.py

```diff
@@ -2582,7 +2584,7 @@ def generate_genlist_for_target(self, genlist: build.GeneratedList, target: buil
         exe = generator.get_exe()
         infilelist = genlist.get_inputs()
         outfilelist = genlist.get_outputs()
-        extra_dependencies = self.get_target_depend_files(genlist)
+        extra_dependencies = self.get_target_depend_files(genlist) # FIXME, function does not accept GeneratedList
```

Comment:
This is something I've noticed while debugging; not directly related to PR itself.